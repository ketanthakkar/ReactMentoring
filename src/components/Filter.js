import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { sortMovies, fetchMovies } from '../actions/index'
import PropTypes from 'prop-types';

class Filter extends Component {

    handleSortClick = (event) => {
        const sortBy = event.target.id === "release" ? "release_date" : "vote_average";
        this.props.sortMovies(sortBy);
        this.props.fetchMovies();

        document.getElementById("release").classList.toggle("highlight-color");
        document.getElementById("rating").classList.toggle("highlight-color");
      };

    render() {
        return (
            <section className="filter-container">
                <span className="results">{this.props.movieCount} movies found</span>
                <div className="sort-detail">
                    <span>Sort by</span>
                    <span id="release" className="highlight-color" onClick={this.handleSortClick}>release date</span>
                    <span id="rating" className="" onClick={this.handleSortClick}>rating</span>
                </div>
            </section>
        )
    }
}

Filter.propTypes = {
    movieCount: PropTypes.number
};

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
      sortMovies,
      fetchMovies
    }, dispatch);
  }

export default connect(null, mapDispatchToProps)(Filter);