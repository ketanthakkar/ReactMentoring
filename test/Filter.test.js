import React from 'react';
import { shallow } from 'enzyme';
import Filter from "../src/components/Filter";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({});

describe('should renders Filter correctly', () => {
  
  test('Snapshot test', () => {
    const tree = shallow(
      <Provider store={store}>
        <Filter movieCount={10}/>
      </Provider>);
    expect(tree).toMatchSnapshot();
  });

});
