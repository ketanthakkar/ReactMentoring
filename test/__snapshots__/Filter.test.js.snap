// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders Filter correctly Snapshot test 1`] = `
<section
  className="filter-container"
>
  <span
    className="results"
  >
    10
     movies found
  </span>
  <div
    className="sort-detail"
  >
    <span>
      Sort by
    </span>
    <span>
      release date
    </span>
    <span>
      rating
    </span>
  </div>
</section>
`;
